chronos/asyncfutures2.nim

```diff
@@ -398,6 +402,39 @@ proc `cancelCallback=`*[T](future: Future[T], cb: CallbackFunc) =
   ## This callback will be called immediately as ``future.cancel()`` invoked.
   future.cancelcb = cb
 
+{.push stackTrace: off.}
+proc futureContinue*[T](fut: Future[T]) {.gcsafe, raises: [Defect].} =
+  # Used internally by async transformation
+  try:
+    if not (fut.closure.finished()):
+      var next = fut.closure(fut)
+      while (not next.isNil()) and
+          next.finished():
+        next = fut.closure(fut)
+        if fut.closure.finished():
+          break
+      if next == nil:
+        if not (fut.finished()):
+          const
+            msg = "Async procedure (&" & "testInner" &
+                ") yielded `nil`, " &
+                "are you await\'ing a `nil` Future?"
+          raiseAssert msg
+      else:
+        next.addCallback(proc (arg: pointer) {.gcsafe, raises: [Defect].} =
+          futureContinue(fut))
+  except CancelledError:
+    fut.cancelAndSchedule()
+  except CatchableError as exc:
+    fut.fail(exc)
+  except Exception as exc:
```

Comment:
This handler MUST be removed. All the `Defect` errors will be raised as `Defect` and crash event loop. `Exception` is parent of `Defect` so any `Exception` MUST crash event loop too.

Reply:
You mean with `-d:chronosStrictException` or all the time?
I think I copied this from the old code: https://github.com/status-im/nim-chronos/blob/master/chronos/asyncmacro2.nim#L60-L99
but maybe I miscopied something

Reply:
I'll copy it again to be sure